sarc/cli/acquire/jobs.py

```diff
@@ -94,18 +96,30 @@ def execute(self) -> int:
 
         for cluster_name in self.cluster_names:
             for date, is_auto in parse_dates(self.dates, cluster_name):
-                try:
-                    print(
-                        f"Acquire data on {cluster_name} for date: {date} (is_auto={is_auto})"
-                    )
-
-                    sacct_mongodb_import(
-                        clusters_configs[cluster_name], date, self.no_prometheus
-                    )
-                    if is_auto:
-                        _dates_set_last_date(cluster_name, date)
-                # pylint: disable=broad-exception-caught
-                except Exception as e:
-                    print(f"Failed to acquire data for {cluster_name} on {date}: {e}")
-                    return 1
+                with using_trace("AcquireJobs", "cluster-data-is_auto") as span:
+                    span.set_attribute("cluster_name", cluster_name)
+                    span.set_attribute("date", str(date))
+                    span.set_attribute("is_auto", is_auto)
+                    try:
+                        span.add_event(
+                            f"Acquire data on {cluster_name} for date: {date} (is_auto={is_auto})"
+                        )
+                        logging.info(
+                            f"Acquire data on {cluster_name} for date: {date} (is_auto={is_auto})"
+                        )
+
+                        sacct_mongodb_import(
+                            clusters_configs[cluster_name], date, self.no_prometheus
+                        )
+                        if is_auto:
+                            _dates_set_last_date(cluster_name, date)
+                    # pylint: disable=broad-exception-caught
+                    except Exception as e:
+                        span.add_event(
+                            f"Failed to acquire data for {cluster_name} on {date}: {e}"
+                        )
+                        logging.error(
+                            f"Failed to acquire data for {cluster_name} on {date}: {e}"
+                        )
+                        return 1
```

Comment:
@nurbal Si ça plante pour une grappe ça exit pour les autres?

Reply:
Oui, il semblerait à cause du return.
Un `raise` serait plus approprié, il serait catché par le `using_trace` et on passerait à la date suivante, mais dans ce cas attention:
Il faudrait même d'ailleurs monter le `using trace` d'un cran et considérer que toutes les dates d'un cluster sont dans le même `using_traces`pour ne pas exécuter `_dates_set_last_date(cluster_name, date)` de la ligne 112 si une date antérieure a crashé, dans ce cas. (sinon gare aux trous dans l'historique)

Reply:
Ah, c'est un for-loop sur les dates! En effet, ça serait mieux de ne pas parser les dates subséquentes. 

Faudrait break la loop pour la grappe pour skipper les autres dates mais au moins passer à la prochaine grappe.

Reply:
Pour garder le même comportement qu'actuellement, on pourrait:
- faire un raise au lieu de `return 1`
- modifier le `using_trace` en ajoutant `exception_types=()`, pour qu'il ne capture aucune exception, et donc les re-raise toutes.

Ainsi, la moindre exception arrêtera l'exécution du code, comme le fait actuellement le `return 1`. En contre-partie, en laissant le `using_trace` à la même place (donc, sans le monter), le span qui recevra une erreur aura les attributs correspondant précisément à ce qui a causé l'erreur (c'est-à-dire le `cluster_name` et la `date`).

Qu'en pensez-vous ? (ou bien, on laisse aussi le code tel qu'il est actuellement, avec le `return 1` ?)

Reply:
PS: Ah, désolé, je n'avais pas vu la réponse de Xavier.

Reply:
Commit ajouté: https://github.com/mila-iqia/SARC/pull/111/commits/a1c38761f394b223fb78d9caed8a53a453620546

Donc, conformément au commentaire de @bouthilx : la loop est break si on échoue pour une date, mais on passe au cluster suivant.